### PR TITLE
New Uma model and animations

### LINKED ISSUE #1276: [REFACTOR] Update Uma model and animations
## 🚧 Refactor

Ditto

### PR DESCRIPTION
**Pull request by BertovDev**
Closes #1276 

## Motivation

New Uma animations 

## Summary of changes

1. Add new Animations and Model
2. Change mainscreen and characterInfo animations display from Victory to Radiance
3. Change `UIModelManager` character animation functionality

## How has this been tested?

To test it add "Uma" To the enableCharacter list in MainScreenManager script

## Test additions / changes

List tests added/updated.

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

File: client/Assets/Scripts/UI/UIModelManager.cs [hunk 1]
```diff
@@ -38,17 +38,22 @@ public void RemoveCurrentModel()
     float AnimationClipTime(Animator modelClone)
     {
         List<AnimationClip> clips = modelClone.runtimeAnimatorController.animationClips.ToList();
-        return clips.Single(clip => clip.name == "Victory").length;
+        return modelClone.name.Contains("Uma")
+            ? clips.Single(clip => clip.name == "GR_VeilOfRadiance").length / 2
```
Comment (agustinesco) [inline client/Assets/Scripts/UI/UIModelManager.cs:42]: Why not renaming the animation "GR_VeilOfRadiance" to "victory" if we want to use it as victory animation? we could also duplicate it

File: client/Assets/Scripts/UI/UIModelManager.cs [hunk 2]
```diff
@@ -38,17 +38,22 @@ public void RemoveCurrentModel()
...
+            : clips.Single(clip => clip.name == "Victory").length;
     }
 
     IEnumerator AnimateCharacter(GameObject modelClone)
     {
+        // Fix this: With the characterSelection PR we can add a string in the ComCharacter to
+        // select which animations should be played
+        string animationName = modelClone.name.Contains("Uma") ? "Radiance" : "Victory";
```
Comment (agustinesco) [inline client/Assets/Scripts/UI/UIModelManager.cs:50]: same as above

**Review by agustinesco — CHANGES_REQUESTED** [review]

**Comment by agustinesco** [comment]
https://github.com/lambdaclass/curse_of_mirra/assets/106101218/6dbdbe19-ee63-4746-b136-dbc22354ab47

While i tested this pr i noticed some issues in the character selection screen, lets work on them on different prs since this one only adds the uma character:
- I think the characters available should be sorted first and the not yet available second
- If i select uma as the screen is taking me to muflus
- Every time i change characters i got an unity exception

**Comment by agustinesco** [comment]
When i enter a match as uma i'm  getting these errors: 
image
image

**Comment by agustinesco** [comment]
and one last thing, the animations for uma doesn't have the flag "Has model animation" as true, so she is not doing it.

for the sake of testing i used this [backend branch](https://github.com/lambdaclass/game_backend/pull/107) to test this pr so i was able to hit

File: client/Assets/Scripts/UI/UIModelManager.cs (hunk 1 above)
Comment (BertovDev) [reply client/Assets/Scripts/UI/UIModelManager.cs:42]: We only want to use the "GR_VeilOfRadiance" as victory animation in the showcase of the characters (mainscreen and characterInfo). In the endgame animation we want to show the `Victory` animation not `GR_VeilOfRadiance`. Finally we also use GR_VeilOfRadiance as the animation for the skill1.

**Comment by BertovDev** [comment]
> Screen.Recording.2023-12-21.at.17.01.28.mov
> 
> While i tested this pr i noticed some issues in the character selection screen, lets work on them on different prs since this one only adds the uma character:
> 
>     * I think the characters available should be sorted first and the not yet available second
> 
>     * If i select uma as the screen is taking me to muflus
> 
>     * Every time i change characters i got an unity exception

This is fixed in #1000

**Comment by BertovDev** [comment]
> When i enter a match as uma i'm getting these errors: image image
> 
> and this one when i receive damage: image

Awesome catch! pushed the fix!

File: client/Assets/Scripts/UI/UIModelManager.cs (hunk 1 above)
Comment (agustinesco) [reply client/Assets/Scripts/UI/UIModelManager.cs:42]: oh okey, maybe in the future we could add a new skill to the model to do in both the characters and the mainscreen right? (not in this pr)

**Comment by agustinesco** [comment]
Great work! we're almost there, i just noticed one more errors that i think we should fix

When any uma dies i got this error
image

**Comment by BertovDev** [comment]
> Great work! we're almost there, i just noticed one more errors that i think we should fix
> 
> When any uma dies i got this error image


Done!

**Review by agustinesco — DISMISSED** [review]
LGTM! great work

**Review by ncontinanza — CHANGES_REQUESTED** [review]
Looking great! I think that the UI Icon for the killfeed is missing, by adding that we're good to go

**Review by ncontinanza — APPROVED** [review]
LGTM! Just notice that until https://github.com/lambdaclass/game_backend/pull/107 is not fixed and merged, it won't be possible to use Uma in a battle.

**Review by agustinesco — APPROVED** [review]
LGTM! tested on local and the model seems great!

**Pull request merged**